src/components/TipControl.vue

```diff
@@ -0,0 +1,234 @@
+
+<template>
+  <a
+    v-if="USE_DEEP_LINKS"
+    :href="deepLink"
+    target="_blank"
+    class="tip__content"
+  >
+    <img :src="iconTip">
+    <ae-amount
+      :amount="tip.retip_amount_ae.toString()"
+      :round="2"
+      class="vertical-align-mid"
+    />
+    <fiat-value
+      :amount="tip.retip_amount_ae.toString()"
+      class="vertical-align-mid"
+    />
+  </a>
+  <div
+    v-else
+    class="tip-url__wrapper"
+    title="Total amount of retips"
+  >
+    <div
+      v-if="show"
+      class="overlay"
+      @click="toggleTip(false)"
+    />
+    <div
+      class="tip__container_wrapper"
+      @:click.stop
+    >
+      <div
+        class="retip__content"
+        :class="[{ active: show }]"
+        @click="toggleTip(!show)"
+      >
+        <img
+          class="tip__icon"
+          :src="iconTip"
+        >
+        <ae-amount
+          :amount="tip.retip_amount_ae"
+          :round="2"
+        />
+        <fiat-value :amount="tip.retip_amount_ae.toString()" />
+      </div>
+      <div
+        v-if="show"
+        class="tip__container"
+      >
+        <loading :show-loading="showLoading" />
+        <div
+          v-show="error && !showLoading"
+          class="text-center mb-2"
+        >
+          An error occurred while sending tip
+        </div>
+        <form
+          v-if="!showLoading"
+          @submit.prevent
+        >
+          <div
+            class="input-group"
+          >
+            <input
+              v-model="message"
+              type="text"
+              class="form-control tip__message"
+              placeholder="Add message"
+            >
+          </div>
+          <div class="amount__row">
+            <ae-input-amount v-model="value" />
```

Comment:
```suggestion
            <ae-input-amount v-model.number="value" />
```

Reply:
same should be to retip

Reply:
Else, there is a validation issue: when you edit the number field and return it to 0, it becomes '0' (as string) and validation starts failing